elearning-frontend/src/components/admin/CreateWords.js

```diff
@@ -0,0 +1,335 @@
+import React, { useCallback, useEffect, useState } from 'react'
+import { useNavigate } from 'react-router-dom'
+import apiClient from '../../services/api'
+import Toastify from '../../core/Toastify'
+import saveWord from '../actions/saveWord'
+import InputField from '../../core/InputField'
+import Divider from '../../core/Divider'
+import { ToastContainer } from 'react-toastify'
+
+const CreateWords = (props) => {
+    const navigate = useNavigate()
+
+    // Cateogry List
+    const [categoryList, setCategoryList] = useState([])
+    const [isCategoryLoading, setIsCategoryLoading] = useState(true)
+
+    // Word
+    const [word, setWord] = useState({
+        category_id: '',
+        word: '',
+        choices: {
+            firstChoice: {
+                choice: '',
+                is_correct: false,
+                label: 'firstChoice'
+            },
+            secondChoice: {
+                choice: '',
+                is_correct: false,
+                label: 'secondChoice'
+            },
+            thirdChoice: {
+                choice: '',
+                is_correct: false,
+                label: 'thirdChoice'
+            },
+            fourthChoice: {
+                choice: '',
+                is_correct: false,
+                label: 'fourthChoice'
+            },
+        }
+    })
+
+    // Fetch all categories
+    const GetCategoryList = useCallback(async () => {
+        apiClient({
+            method: "get",
+            url: "/api/admin/categories"
+        }).then(response => {
+            setCategoryList(response.data.categories)
+            setIsCategoryLoading(false)
+        }).catch(error => {
+            Toastify("error", error)
+        })
+    }, [])
+
+    useEffect(() => {
+        if (!props.session) {
+            navigate('/')
+        }
+        else {
+            if(props.admin) {
+                GetCategoryList()
+            }
+            else {
+                navigate('/dashboard')
+            }
+        }
+    }, [props.session])
```

Comment:
- [ ] separate the session checker, it should be checked at app.js or index.js level